packages/verkle/src/node/internalNode.ts

```diff
@@ -1,119 +1,91 @@
-import { equalsBytes } from '@ethereumjs/util'
-
-import { POINT_IDENTITY } from '../util/crypto.js'
+import { type VerkleCrypto } from '@ethereumjs/util'
 
 import { BaseVerkleNode } from './baseVerkleNode.js'
-import { LeafNode } from './leafNode.js'
 import { NODE_WIDTH, VerkleNodeType } from './types.js'
 
-import type { VerkleNode, VerkleNodeOptions } from './types.js'
+import type { ChildNode, VerkleNodeOptions } from './types.js'
 
 export class InternalNode extends BaseVerkleNode<VerkleNodeType.Internal> {
-  // Array of references to children nodes
-  public children: Array<VerkleNode | null>
-  public copyOnWrite: Record<string, Uint8Array>
+  // Array of tuples of uncompressed commitments (i.e. 64 byte Uint8Arrays) to child nodes along with the path to that child (i.e. the partial stem)
+  public children: Array<ChildNode>
   public type = VerkleNodeType.Internal
 
-  /* TODO: options.children is not actually used here */
   constructor(options: VerkleNodeOptions[VerkleNodeType.Internal]) {
     super(options)
-    this.children = options.children ?? new Array(NODE_WIDTH).fill(null)
-    this.copyOnWrite = options.copyOnWrite ?? {}
-  }
-
-  commit(): Uint8Array {
-    throw new Error('Not implemented')
-  }
-
-  cowChild(_: number): void {
-    // Not implemented yet
+    this.children =
+      options.children ??
+      new Array(256).fill({
+        commitment: options.verkleCrypto.zeroCommitment,
+        path: new Uint8Array(),
+      })
   }
 
-  setChild(index: number, child: VerkleNode) {
-    this.children[index] = child
+  // Updates the commitment value for a child node at the corresponding index
+  setChild(childIndex: number, child: ChildNode) {
+    // Get previous child commitment at `index`
+    const oldChildReference = this.children[childIndex]
+    // Updates the commitment to the child node at `index`
+    this.children[childIndex] = { ...child }
```

Comment:
Why are we spreading into an object? To clone the object and not just copy the reference?

Reply:
Correct.  Because objects are reference types in Javascript, I need to clone it here because the `child` was being passed by reference in the call to `setChild` and then getting garbage collected when `setChild` returned.  I think this is probably essentially what the `copyOnWrite` functionality is doing in `go-verkle`